lib/rubocop/cop/performance/string_replacement.rb

```diff
@@ -86,7 +86,8 @@ def accept_first_param?(first_param)
 
           unless first_param.str_type?
             return true if options
-            return true unless first_source =~ DETERMINISTIC_REGEX
+            return true unless first_source.is_a?(String) &&
```

Comment:
I think it's better to use Regexp#match

```
DETERMINISTIC_REGEX.match(first_source)
```

Regexp#match? does not implemented on jruby

Reply:
Hmm, It looks like there is no advantage to replacing it with `Regexp#match`.

```diff
% git diff
diff --git a/lib/rubocop/cop/performance/string_replacement.rb b/lib/rubocop/cop/performance/string_replacement.rb
index 848180ec0..1ca8e9fe8 100644
--- a/lib/rubocop/cop/performance/string_replacement.rb
+++ b/lib/rubocop/cop/performance/string_replacement.rb
@@ -86,8 +86,7 @@ module RuboCop

           unless first_param.str_type?
             return true if options
-            return true unless first_source.is_a?(String) &&
-                               first_source =~ DETERMINISTIC_REGEX
+            return true unless DETERMINISTIC_REGEX.match(first_source)

             # This must be done after checking DETERMINISTIC_REGEX
             # Otherwise things like \s will trip us up
```

This change will cause the following `TypeError`.

```console
% bundle exec rspec spec/rubocop/cop/performance/string_replacement_spec.rb

(snip)

Failures:

  1) RuboCop::Cop::Performance::StringReplacement non deterministic regex allows regex literal containing interpolations
     Failure/Error: return true unless DETERMINISTIC_REGEX.match(first_source)

     TypeError:
       no implicit conversion of RuboCop::AST::Node into String
     # ./lib/rubocop/cop/performance/string_replacement.rb:89:in `match'
     # ./lib/rubocop/cop/performance/string_replacement.rb:89:in `accept_first_param?'
     # ./lib/rubocop/cop/performance/string_replacement.rb:40:in `block in on_send'
     # ./lib/rubocop/cop/performance/string_replacement.rb:47:in `string_replacement?'
     # ./lib/rubocop/cop/performance/string_replacement.rb:38:in `on_send'
     # ./spec/rubocop/cop/performance/string_replacement_spec.rb:214:in `block (3 levels) in <top (required)>'

  2) RuboCop::Cop::Performance::StringReplacement non deterministic regex allows regex constructor containing regex with interpolations
     Failure/Error: return true unless DETERMINISTIC_REGEX.match(first_source)

     TypeError:
       no implicit conversion of RuboCop::AST::Node into String
     # ./lib/rubocop/cop/performance/string_replacement.rb:89:in `match'
     # ./lib/rubocop/cop/performance/string_replacement.rb:89:in `accept_first_param?'
     # ./lib/rubocop/cop/performance/string_replacement.rb:40:in `block in on_send'
     # ./lib/rubocop/cop/performance/string_replacement.rb:47:in `string_replacement?'
     # ./lib/rubocop/cop/performance/string_replacement.rb:38:in `on_send'
     # ./spec/rubocop/cop/performance/string_replacement_spec.rb:228:in `block (3 levels) in <top (required)>'

Finished in 0.14404 seconds (files took 1.53 seconds to load)
133 examples, 2 failures
```

I think it would be better to follow upstream changes.
ruby/ruby@f7e94df

Reply:
I see. Thanks to consider it.